decision_service/dsclientlib/ds_driver.cc

```diff
@@ -0,0 +1,131 @@
+#include "ds_driver.h"
+#include "ds_event.h"
+#include "logger/ds_logger.h"
+
+#include <boost/uuid/uuid_generators.hpp> // generators
+#include <boost/uuid/uuid_io.hpp>         // streaming operators etc.
+#include "error_callback_fn.h"
+
+
+//this macro assumes that success_code equals 0
+#define TRY_OR_RETURN(x) do { \
+  int retval = (x); \
```

Comment:
>retval [](start = 6, length = 6)

add __LINENR__ (or whatever it's called to make sure the variable name is unique

Reply:
retval is scoped by the do {} while block.  Is there a reason to make it unique?

---
In reply to: [186734938](https://github.com/JohnLangford/vowpal_wabbit/pull/1479#discussion_r186734938) [](ancestors = 186734938)

Reply:
if I have a retval variable in the surrounding block it's clashing.

---
In reply to: [186922880](https://github.com/JohnLangford/vowpal_wabbit/pull/1479#discussion_r186922880) [](ancestors = 186922880,186734938)

Reply:
makes sense

---
In reply to: [187031419](https://github.com/JohnLangford/vowpal_wabbit/pull/1479#discussion_r187031419) [](ancestors = 187031419,186922880,186734938)